pygqlc/GraphQLClient.py

```diff
@@ -346,7 +349,25 @@ def _start(self, payload, _id=None):
   def _stop(self, _id):
     payload = {'id': _id, 'type': 'stop'}
     self._conn.send(json.dumps(payload))
-  
+
+  def resetSubsConnection(self):
+    if not self.sub_router_thread:
+      print('connection not stablished, nothing to reset')
+      return False
+    if self.sub_router_thread.isAlive(): #check that _sub_routing_loop() is running
+      self.wss_conn_halted = True 
+    else: # in case for some reason _sub_routing_loop() is not running 
+      if self._new_conn():
+        print('WSS Reconnection succeeded, attempting resubscription to lost subs')
+        self._resubscribe_all()
+        print('finished resubscriptions')
+        self.sub_router_thread = threading.Thread(target=self._sub_routing_loop)
+        self.sub_router_thread.start() #start again _sub_routing_loop()
+      else:
+        print('Reconnection has not been possible')
+        return False
+    return True
```

Comment:
Se ve bien, solo reordenaría un poco porque están organizados algo chistoso los if/else (y el último return True, que no se ve a simple vista que caso lo genera):
```suggestion
  def resetSubsConnection(self):
    if not self.sub_router_thread:
      print('connection not stablished, nothing to reset')
      return False
    if self.sub_router_thread.isAlive(): #check that _sub_routing_loop() is running
      self.wss_conn_halted = True
      return True
    # in case for some reason _sub_routing_loop() is not running 
    if self._new_conn():
      print('WSS Reconnection succeeded, attempting resubscription to lost subs')
      self._resubscribe_all()
      print('finished resubscriptions')
      self.sub_router_thread = threading.Thread(target=self._sub_routing_loop)
      self.sub_router_thread.start() #start again _sub_routing_loop()
      return True
    else:
      print('Reconnection has not been possible')
      return False
```

Reply:
mas que nada junté el return True con el código que lo genera, y eliminé la anidación de if / else para seguir el código mas facil

Reply:
tienes razón, de esa manera se entiende mejor 👍